src/DominantJuice.sol

```diff
@@ -0,0 +1,396 @@
+    error NoRefundsForSuccessfulCycle();
+    error ContractAlreadyInitialized();
+    error AlreadyWithdrawnRefund();
+    error InvalidRedemptionEvent(address caller, uint256 projectId, uint256 value);
+    error InsufficientFunds();
+
+    /// @notice Indicates if this contract adheres to the specified interface.
+    /// @dev See {IERC165-supportsInterface}.
+    /// @param _interfaceId The ID of the interface to check for adherence to.
+    /// @return A flag indicating if the provided interface ID is supported.
+    function supportsInterface(bytes4 _interfaceId) public view virtual override returns (bool) {
+        return _interfaceId == type(IJBFundingCycleDataSource3_1_1).interfaceId
+            || _interfaceId == type(IJBPayDelegate3_1_1).interfaceId
+            || _interfaceId == type(IJBRedemptionDelegate3_1_1).interfaceId;
+    }
+
+    /// @notice Initializes the Dominant Assurance cloned contract with project details
+    /// and the pertinent Juicebox architecture contracts. Function should be called by the
+    /// Delegate Deployer contract directly after it creates the clone for each new project.
+    /// @param _projectId The ID of the project this contract's functionality applies to.
+    function initialize(
+        uint256 _projectId,
+        uint256 _cycleTarget,
+        uint256 _minimumPledgeAmount,
+        IJBController3_1 _controller,
+        IJBSingleTokenPaymentTerminalStore3_1_1 _paymentTerminalStore
+    ) external {
+        // Stop re-initialization.
+        if (projectId != 0) revert ContractAlreadyInitialized();
+
+        // Store project parameters and ready JB architecture contracts for calling
+        cycleTarget = _cycleTarget;
+        minimumPledgeAmount = _minimumPledgeAmount;
+        projectId = _projectId;
+        controller = _controller;
+        paymentTerminalStore = _paymentTerminalStore;
+        directory = controller.directory();
+        fundAccessConstraintsStore = controller.fundAccessConstraintsStore();
+        address terminalAddr = address(directory.terminalsOf(projectId)[0]);
+        paymentTerminal = IJBSingleTokenPaymentTerminal(terminalAddr);
+        //paymentToken = paymentTerminal.token();
+        (JBFundingCycle memory cycleData,) = controller.currentFundingCycleOf(projectId);
+        cycleExpiryDate = cycleData.start + cycleData.duration;
+    }
+
+    /**
+     * @notice Project creator calls this function to deposit refund bonus before funding cycle start
+     * to increase potential pledger confidence.
+     * @dev Can only be called by the contract owner / project creator.
+     */
+    function depositRefundBonus(uint256 refundBonusAmount) external payable onlyOwner {
+        // Reverts if initialize() has not been called yet.
+        if (projectId == 0) revert DataSourceNotInitialized();
+
+        // Reverts if ETH sent does not match input amount.
+        if (msg.value != refundBonusAmount) revert FundsMustMatchInputAmount(refundBonusAmount);
```

Comment:
why the `refundBonusAmount` parameter at all if this is always the same as the sent ETH? seems like either the sent ETH can be more than the refundBonusAmount OR the param is redundant

Reply:
I had put this in as an extra check for whoever was calling this function to try to encourage intentional inputs, since if it's an incorrect amount based on a typo from the user, the money sent will be locked up for around a month.... I think I'm okay taking it out. We just need to tell the multisig to be extra careful when calling that function on game day.